### ISSUE TITLE
[BUG][Fuzzer][wrong-code] `_tir_f32x2_to_bf16x2_to_u32` silently turns inf/NaN into ±0.0/±inf instead of preserving it — RNE bias add has no `exp == 0xFF` guard

### ISSUE THREAD

**Issue by Hughshine** [tile-ai/tilelang#2481]
### Required prerequisites

- [x] I have read the documentation <https://tilelang.com>.
- [x] I have searched the [Issue Tracker](https://github.com/tile-ai/tilelang/issues) that this hasn't already been reported. (comment there if it has.)

### What version of TileLang are you using?

0.1.11

### System information

- Installed from the `0.1.11` wheel; source pinned to tag `v0.1.11` (commit `cd37ed5fc35ae7a60a1277c8eb49028174ac51e6`).
- Python 3.13, linux. GPU: NVIDIA L40S (sm_89). The defect is in a host-independent TIR convert.

### Problem description

`_tir_f32x2_to_bf16x2_to_u32` packs two f32 into a bf16x2-in-uint32 with round-to-nearest-even. The RNE step adds a rounding bias to the f32 bit pattern **without first checking for the inf/NaN exponent (`0xFF`)**. For an inf/NaN input whose discarded low-16 mantissa bits make the bias add carry up, the carry propagates through the exponent and into the sign bit, so the result is no longer inf/NaN: e.g. a NaN `0x7FFF8C22` becomes bf16 `0x8000` = **-0.0**. A correct bf16 RNE leaves inf/NaN unrounded. NaN *is* representable in bf16 (it should round-trip to NaN), so this is a value corruption, not precision loss.

All **finite** values round-trip correctly (verified by a 331k-input fuzz incl. an exhaustive rounding-boundary sweep — zero finite-value defects); the corruption is confined to inf/NaN inputs.

<details>
<summary>Root cause — RNE bias add with no <code>exp == 0xFF</code> guard</summary>

[`quantization.py#L96-L98`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/tilelang/quantize/quantization.py#L96-L98):

```python
rounding_bias = ((u32_val >> 16) & 1) + 0x7FFF
u32_val += rounding_bias          # <-- no `if exp == 0xFF: skip` guard
# bf16 = (u32_val >> 16) & 0xFFFF
```

For `0x7FFF8C22` the guard bit is 1 so `bias = 0x8000`; `0x7FFF8C22 + 0x8000 = 0x80000C22`, whose top 16 bits are `0x8000` = bf16 -0.0. Depending on the mantissa the carry can instead yield `0x7F80` (→ +inf) or wrap to `0x0000` (→ +0.0). Standard bf16 RNE (e.g. the one in upstream frameworks) special-cases `exp == 0xFF` to pass inf/NaN through unrounded; this codec omits it.

### Observed vs expected (independently re-derived; pure-host replay matches silicon bit-for-bit)

| f32 input (NaN) | expected (bf16 RNE) | this codec |
|---|---|---|
| `0x7FFF8C22` | NaN | **-0.0** |
| `0xFFFFA91E` | NaN | **+0.0** |
| `0x7FFFFFFF` | NaN | **-0.0** |
| `0x7FC00000` (canonical qNaN) | NaN | NaN (ok — bias add doesn't carry) |
</details>

### Reproducible example code

Note: use a real `.py` file; `tilelang.disable_cache()`.

```python
import tilelang, tilelang.language as T, torch
tilelang.disable_cache()
from tilelang.quantize.quantization import _tir_f32x2_to_bf16x2_to_u32 as ENC, _tir_u32_to_bf16x2_to_f32x2 as DEC

@T.prim_func
def main(I: T.Tensor((2,), "uint32"), O: T.Tensor((2,), "float32")):
    with T.Kernel(1, threads=1) as _:
        f = T.alloc_local((2,), "float32")
        f[0] = T.reinterpret("float32", I[0]); f[1] = T.reinterpret("float32", I[1])
        w = ENC(f[0], f[1])
        d = list(DEC(w))
        O[0] = d[0]; O[1] = d[1]

k = tilelang.compile(main, target="cuda")
I = torch.tensor([0x7FFF8C22, 0x7FFFFFFF], dtype=torch.uint32, device="cuda")  # two NaNs
O = torch.zeros(2, dtype=torch.float32, device="cuda")
k(I, O); torch.cuda.synchronize()
print(O.cpu().tolist())   # [-0.0, -0.0]  — expected [nan, nan]
```

### Traceback

```pytb
No traceback — the pack compiles and runs to completion; an inf/NaN input is silently turned into a finite ±0.0/±inf, deterministically.
```

### Expected behavior

The RNE pack should leave inf/NaN (`exp == 0xFF`) unrounded so they round-trip to bf16 inf/NaN, OR `_tir_f32x2_to_bf16x2_to_u32` should be documented/guarded as finite-only. Producing ±0.0/±inf from a NaN is a silent value corruption that breaks NaN propagation.

### Additional context

**Suggested fix (proposed, not built).** Guard the bias add on the exponent: only add `rounding_bias` when `(u32_val >> 23) & 0xFF != 0xFF` (i.e. skip rounding for inf/NaN), matching the standard f32→bf16 RNE.

Confirmed on an L40S (sm_89, 0.1.11) by round-tripping through the real exported functions and re-derived with a pure-host replay that matches the silicon output bit-for-bit; I did not build the fix. `_tir_f32x2_to_bf16x2_to_u32` has zero in-library callers — it is exported/user-callable, so this affects a user calling the pack on inf/NaN data. All finite values round-trip correctly (331k-input fuzz incl. an exhaustive rounding-boundary sweep, zero finite-value defects), so the corruption is confined to inf/NaN.
